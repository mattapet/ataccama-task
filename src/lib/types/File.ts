/**
 * These types describe shape of the expected data.
 */

export interface Data {
  [key: string]: string;
}

export interface Kid {
  records: Record[];
}

export interface Kids {
  [key: string]: Kid;
}

export interface Record {
  _id: string;
  data: Data;
  kids: Kids;
}

export type Records = Record[];

/**
 * Root file representation.
 */
export type File = Record[];
