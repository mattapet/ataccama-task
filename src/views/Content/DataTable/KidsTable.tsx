//#region imports
import React from 'react';
import { Kids } from 'lib/types/File';
import RecordsTable from './RecordsTable';
//#endregion

//#region Component interfaces
export interface Props {
  /**
   * Kids to render.
   */
  readonly data: Kids;
}
//#endregion

/**
 * Component that renders a `RecordTable` for each of the kids.
 */
const KidsTable: React.FunctionComponent<Props> = props => (
  <React.Fragment>
    {Object.keys(props.data).map(subtable => (
      <RecordsTable
        key={subtable}
        title={subtable}
        records={props.data[subtable].records}
      />
    ))}
  </React.Fragment>
);

export default KidsTable;
