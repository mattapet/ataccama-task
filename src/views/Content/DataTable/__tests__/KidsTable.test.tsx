//#region imports
import React from 'react';
import renderer from 'react-test-renderer';

import KidsTable from '../KidsTable';
import { Kids } from 'lib/types/File';
//#endregion

it('should render correctly', () => {
  // Arrange
  const kids1: Kids = {};
  const kids2: Kids = {
    test: {
      records: [
        {
          _id: 'f',
          data: { a: '2', c: '4' },
          kids: {},
        },
      ],
    },
  };

  // Act
  const component1 = renderer.create(<KidsTable data={kids1} />);
  const component2 = renderer.create(<KidsTable data={kids2} />);

  // Assert
  expect(component1).toMatchSnapshot();
  expect(component2).toMatchSnapshot();
});
