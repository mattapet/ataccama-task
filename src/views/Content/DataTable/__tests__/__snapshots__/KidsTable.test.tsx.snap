// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `null`;

exports[`should render correctly 2`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-default ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-title"
        >
          <h3>
            test
          </h3>
        </div>
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  className="ant-table-expand-icon-col"
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className="ant-table-expand-icon-th"
                    rowSpan={1}
                    title=""
                  />
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          a
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          c
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Actions
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="f"
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className="ant-table-row-expand-icon-cell"
                  />
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    2
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    4
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <a
                      href="javascript:;"
                      onClick={[Function]}
                    >
                      Remove
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
